classes/Product.php

```diff
@@ -3379,6 +3379,43 @@ public static function priceCalculation(
             $real_quantity
         );
 
+        // If cart has been ordered the price is saved in OrderDetails
+        // When we want only the reduction no need for this
+        if ($id_cart && !$only_reduc) {
+            $orderId = Order::getIdByCartId($id_cart);
+            if (false !== $orderId) {
+                $sql = new DbQuery();
+                $sql->select('od.*, t.rate AS tax_rate');
+                $sql->from('order_detail', 'od');
+                $sql->where('od.`id_order` = ' . (int) $orderId);
+                $sql->where('od.`product_id` = ' . (int) $id_product);
+                if (Combination::isFeatureActive()) {
+                    $sql->where('od.`product_attribute_id` = ' . (int) $id_product_attribute);
+                }
+                $sql->leftJoin('order_detail_tax', 'odt', 'odt.id_order_detail = od.id_order_detail');
+                $sql->leftJoin('tax', 't', 't.id_tax = odt.id_tax');
+                $res = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS($sql);
+                if (is_array($res) && count($res)) {
```

Comment:
```suggestion
                if (!empty($res)) {
```

Reply:
I was ok at first, but then I checked at all the potential types returned by `executeS` => `array|false|mysqli_result|PDOStatement|resource|null`

I'm not 100% sure I wouldn't have false positive on some of the returned type with just `isset` or `empty` So, since this method is quite critical, out of safety I prefer to use `if (is_array($res) && !empty($res)) {` to make sure I do have an array